<?xml version="1.0"?>
<launch> 

<!-- AirSim ROS Wrapper -->
	<include file="$(find airsim_ros_pkgs)/launch/airsim_node.launch"/>
<!-- Vehicle Dynamic Constraints -->
	<include file="$(find airsim_ros_pkgs)/launch/dynamic_constraints.launch"/>
<!-- Simple PID Position Controller -->
    <node name="pid_position_node_Drone1" pkg="airsim_ros_pkgs" type="pd_position_controller_simple_node" output="screen">
	<param name="vehicle_name" value="Drone1"/>
        <param name="update_control_every_n_sec" type="double" value="0.01" /> <!-- this needs to be in sync with airsim_ros_wrapper? -->
        <param name="kp_x" type="double" value="1.0" />
        <param name="kp_y" type="double" value="1.0" />
        <param name="kp_z" type="double" value="1.0" />
        <param name="kp_yaw" type="double" value="1.0" />

        <param name="kd_x" type="double" value="0.05" />
        <param name="kd_y" type="double" value="0.05" />
        <param name="kd_z" type="double" value="0.05" />
        <param name="kd_yaw" type="double" value="0.05" />

        <param name="reached_thresh_xyz" type="double" value="0.1" />
        <param name="reached_yaw_degrees" type="double" value="5.0" />
    </node>
    <node name="pid_position_node_Drone2" pkg="airsim_ros_pkgs" type="pd_position_controller_simple_node" output="screen">
	<param name="vehicle_name" value="Drone2"/>
        <param name="update_control_every_n_sec" type="double" value="0.01" /> <!-- this needs to be in sync with airsim_ros_wrapper? -->
        <param name="kp_x" type="double" value="1.0" />
        <param name="kp_y" type="double" value="1.0" />
        <param name="kp_z" type="double" value="1.0" />
        <param name="kp_yaw" type="double" value="1.0" />

        <param name="kd_x" type="double" value="0.05" />
        <param name="kd_y" type="double" value="0.05" />
        <param name="kd_z" type="double" value="0.05" />
        <param name="kd_yaw" type="double" value="0.05" />

        <param name="reached_thresh_xyz" type="double" value="0.1" />
        <param name="reached_yaw_degrees" type="double" value="5.0" />
    </node>
</launch>
